apis/configwatch.go

```diff
@@ -181,3 +208,103 @@ func MonitorTLS(start time.Time, expiry time.Time, conf *config.Config) {
 		}
 	}
 }
+
+func (s *Server) getClientValidator(helloInfo *tls.ClientHelloInfo) func([][]byte, [][]*x509.Certificate) error {
+
+	log.Debug().Msgf("Inside get client validator - %v", helloInfo.Conn.RemoteAddr())
+	return func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+		// Verifying client certs with root ca
+		opts := x509.VerifyOptions{
+			Roots:         s.CaCertPool,
+			CurrentTime:   time.Now(),
+			Intermediates: x509.NewCertPool(),
+			KeyUsages:     []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
+		}
+		_, err := verifiedChains[0][0].Verify(opts)
+		if err != nil {
+			log.Error().Err(err).Msgf("certs verification failed")
+			return err
+		}
+
+		log.Debug().Msgf("validating with SAN match rules - %s", s.SANMatchRules)
+		if len(s.SANMatchRules) == 0 {
+			log.Info().Msgf("mtls san match rules are undefined")
+			return nil
+		}
+		for _, dnsName := range verifiedChains[0][0].DNSNames {
+			dnsName := toLowerCaseASCII(dnsName)
+			for _, sanMatchRule := range s.SANMatchRules {
+				sanMatchRule = toLowerCaseASCII(sanMatchRule)
```

Comment:
I think we should remove lower casing of regex patterns and just make sure we mention all matches are lowercase or force case insensitive matches on the supplied regexes '(?i)'.  I think I am ok with just mentioning this in the documentation.